### PR TITLE
Implemented action logics

### PR DESCRIPTION
**Pull request by T00L00**
- Implemented logic for Attack, Dropoff, Gather, and Move actions
- Adjusted Inventory and Unit class as needed to accommodate action logics
- InitMoveAction() in ActionRunner now spawns destination point instead of having spawn marker as child of unit object.
- Resolved merged conflicts with dev
- Created separate scenes for gameplay logic testing and RL testing

### CODE REVIEW

File: Assets/Scripts/SethRefactor/ActionLogics/Move.cs [hunk 1]
```diff
@@ -7,24 +7,31 @@ namespace ScavengerWorld
...
     public class Move : ActionLogic
     {
-        public override bool RequiresInRange(Unit agent, Interactable target)
+        public override bool RequiresInRange(Unit unit, Interactable target)
         {
             return false;
         }
 
-        public override void StartAction(Unit agent, Interactable target)
+        public override void StartAction(Unit unit, Interactable target)
```
Comment (logar16) [inline Assets/Scripts/SethRefactor/ActionLogics/Move.cs:15]: Why does moving require an interactable target?  Why not just a location?

File: Assets/Scripts/SethRefactor/Unit.cs [hunk 2]
```diff
@@ -53,8 +59,33 @@ void Update()
 
         }
 
+        public void Attack(Damageable enemy)
+        {
+            // play little animation
```
Comment (logar16) [inline Assets/Scripts/SethRefactor/Unit.cs:64]: Or particle effect of sparks flying

File: Assets/Scripts/SethRefactor/ActionLogics/DropOff.cs [hunk 3]
```diff
@@ -7,22 +7,23 @@ namespace ScavengerWorld
...
+        public override bool RequiresInRange(Unit unit, Interactable target)
         {
             return true;
         }
 
-        public override void StartAction(Unit agent, Interactable target)
+        public override void StartAction(Unit unit, Interactable target)
         {
-            Debug.Log("Started Dropoff action!");
+            target.Unit.AddItem(unit.RemoveAllItems());
```
Comment (logar16) [inline Assets/Scripts/SethRefactor/ActionLogics/DropOff.cs:17]: This works for now, but what would you do if they could separately hold food and an item (e.g., sword)?  We wouldn't want to dump everything in one fell swoop!

File: Assets/Scripts/SethRefactor/ActionLogics/Move.cs (hunk 1 above)
Comment (logar16) [reply Assets/Scripts/SethRefactor/ActionLogics/Move.cs:15]: Oh it looks like this particular action is for moving to an interactable, whereas the `Mover` class can just call go to point on the navmesh.  Got it.

**Review by logar16 — APPROVED** [review]
Looks good

File: Assets/Scripts/SethRefactor/ActionLogics/DropOff.cs (hunk 3 above)
Comment (T00L00) [reply Assets/Scripts/SethRefactor/ActionLogics/DropOff.cs:17]: Yeah this is just a very simple prototype inventory mechanic. When we actually have distinct items, a more complex inventory system will have to be made.

File: Assets/Scripts/SethRefactor/ActionLogics/Move.cs (hunk 1 above)
Comment (T00L00) [reply Assets/Scripts/SethRefactor/ActionLogics/Move.cs:15]: Yeah this is a hacky solution for now. Ideally we would  not have a Move action at all.

**Pull request merged**